xbmc/music/MusicDatabase.cpp

```diff
@@ -8579,7 +8574,55 @@ void CMusicDatabase::UpdateTables(int version)
                 "WHERE song.idAlbum = album.idAlbum) "
                 "WHERE EXISTS (SELECT 1 FROM song WHERE song.idAlbum = album.idAlbum)");
   }
-
+  if (version < 82)
+  {
+    // Update artist table combining fanart URL data into strImage field
+    // Clear empty URL data <fanart /> and <thumb />
+    m_pDS->exec("UPDATE artist SET strFanart = '' WHERE strFanart = '<fanart />'");
+    m_pDS->exec("UPDATE artist SET strImage = '' WHERE strImage = '<thumb />'");
+    //Prepare strFanart - strip <fanart>...</fanart>, add aspect to the URLs
+    m_pDS->exec("UPDATE artist SET strFanart = REPLACE(strFanart, '<fanart>', '')");
```

Comment:
This top-level `fanart` element can have an attribute `url` that should be prepended to the URLs in the contained `thumb` elements. XML scrapers can generate this, but it looks like not Python.

Reply:
Do you have an example of the xml produced? Even from Universal Artist scraper (xml) data I have not come across this, and trying to read the regexp it isn't obvious to me either.

Reply:
```xml
<fanart url="http://www.thetvdb.com/banners/">
	<thumb dim="1920x1080" preview="_cache/fanart/original/329083-2.jpg">fanart/original/329083-2.jpg</thumb>
	<thumb dim="1920x1080" preview="_cache/fanart/original/329083-3.jpg">fanart/original/329083-3.jpg</thumb>
</fanart>
```

Reply:
Having checked the xml artist scrapers they have never used `url` or any other attribute so there is nothing needed for this conversion of artist fanart. 

However as you show the xml TV show scrapers do produce top-level fanart element with this attribute, so it will need to be handled in any equivalent changes to fanart in the video library.